### ISSUE TITLE
Problem building wheel using Intel compilers

### ISSUE THREAD

**Issue by gurudave** [numpy/numpy#19703]
I'm attempting to build numpy 1.21.2 for Python 3.8.11 using the Intel compiler suite (2019 update 5). I am able to build numpy and the wheel successfully using the following commands:

```
python setup.py build --compiler=intelem --fcompiler=intelem -j14
python setup.py bdist_wheel
```

However, once I install the resulting wheel and try to import numpy, I get an error about the symbol `__intel_sse2_strchr` being undefined when attempting to import `mtrand.cpython-38-x86_64-linux-gnu.so`. This symbol is present in the relevant Intel compiler libraries, but on further investigation what I discovered is that mtrand.so seems to be getting relinked using gcc during the `bdist_wheel` step and is somehow inconsistent with something else. Inspection with ldd after running the `build` step above indicates it's correctly linked to the Intel libs, including the one containing the "missing" symbol. After `bdist_wheel` is executed, ldd no longer reports mtrand.so being linked to any of the Intel libs.

I'm unsure of quite why this is occurring or how best to keep the `so`s linked to the Intel libs, so any guidance or a suggestion about where to look to control this step would be much appreciated.

**Comment by debarun1234** [comment]
hey can i have a look to this, i have intel compiler suite (2019) maybe i could be of some help

**Comment by gurudave** [comment]
I figured it out. It looks like the build step that tries to detect CPU optimizations was silently failing in the `build` step, so `bdist_wheel` was having another go with gcc instead of Intel. In any case, CPU specific optimizations would probably cause me hassle further down the line, so I've disabled them entirely. The following commands work well for me, maybe they will help someone else.

```
python setup.py build_ext --compiler=intelem --fcompiler=intelem --disable-optimization -j14
python setup.py build --compiler=intelem --fcompiler=intelem --disable-optimization -j14
python setup.py bidst_wheel
```

Presumably it would be fine to roll the `build_ext` and `build` steps together, but I didn't test that yet.
